src/core/note_app.dart

```diff
@@ -1,6 +1,127 @@
+import 'dart:io';
+import 'Note.dart';
+ String? delNoteV, editNoteV, toEdit;
+ 
 class NoteApp {
-  // TODO: Complete the run function
+  
   void run() {
-    print('Hello, world!');
+    
+    
+      int operation ;
+    do { 
+        print(" \nNOTE TAKING APP \n \n ***************** \n 1- Create a note \n 2- Edit a note \n 3- Delete a note \n 4- Print notes \n 5- Search for a note \n 6- Exit \n ***************** \n ");
+    
+      print ("Choose an operation ");
+     operation = int.parse(stdin.readLineSync()!); 
+     if (operation>0 && operation <7){
+    switch (operation) {
+     
+      case 1:
+       //create a note
+    try{
+      String? newNoteTitle;
+      String? newNoteContent;
+         
+      print("Title: \n ****");
+      newNoteTitle = stdin.readLineSync();
```

Comment:
check if the title already exists before adding the note or if the title is empty.